kernel/include/ksched.h

```diff
@@ -144,6 +124,44 @@ static inline int _is_higher_prio_than_current(struct k_thread *thread)
 	return _is_t1_higher_prio_than_t2(thread, _current);
 }
 
+#ifdef CONFIG_MULTITHREADING
+static inline int _is_valid_prio(int prio, void *entry_point)
+{
+	if (prio == K_IDLE_PRIO && _is_idle_thread(entry_point)) {
+		return 1;
+	}
+
+	if (_is_prio_lower_or_equal(prio,
+				    K_LOWEST_APPLICATION_THREAD_PRIO)) {
+		return 0;
+	}
+
+	if (_is_prio_higher_or_equal(prio,
```

Comment:
This makes prio == K_HIGHEST_APPLICATION_THREAD_PRIO return false, but K_HIGHEST_APPLICATION_THREAD_PRIO is a valid priority value.